activerecord/lib/active_record/associations/builder/has_one.rb

```diff
@@ -26,5 +32,32 @@ def self.define_validations(model, reflection)
         model.validates_presence_of reflection.name, message: :required
       end
     end
+
+    def self.touch_record(o, changes, name, touch, touch_method) # :nodoc:
+      record = o.send name
+      if record && record.persisted?
```

Comment:
I think it's better to use a guard clause here instead of a single `if` (after every if comes an else) `return unless record && record.persisted?`

Reply:
Fixed it. Thanks, @azbshiri